src/metabase/api/embed.clj

```diff
@@ -275,20 +280,26 @@
                                           (into {})))))
 
 (defn- remove-locked-parameters [dashboard embedding-params]
-  (let [params-to-remove (get-params-to-remove dashboard embedding-params)
-        param-ids-to-remove (set (for [parameter (:parameters dashboard)
-                                       :when     (contains? params-to-remove (keyword (:slug parameter)))]
-                                   (:id parameter)))
-        linked-field-ids (set (mapcat (params/get-linked-field-ids (:dashcards dashboard)) param-ids-to-remove))
-        remove-parameters (fn [dashcard]
-                            (update dashcard :parameter_mappings
-                                    (fn [param-mappings]
-                                      (remove (fn [{:keys [parameter_id]}]
-                                                (contains? param-ids-to-remove parameter_id)) param-mappings))))]
+  (let [{params-to-remove :remove
+         params-to-keep   :keep}  (classify-params-as-keep-or-remove (:parameters dashboard) embedding-params)
+        param-ids-to-remove       (set (for [parameter (:parameters dashboard)
+                                             :when     (contains? params-to-remove (keyword (:slug parameter)))]
+                                         (:id parameter)))
+        param-ids-to-keep         (set (for [parameter (:parameters dashboard)
+                                             :when     (contains? params-to-keep (keyword (:slug parameter)))]
+                                         (:id parameter)))
```

Comment:
what are your thoughts on using `keep` instead of `for` here?

e.g.
```
(set (keep
      (fn [{:keys [slug id]}]
        (when (contains? params-to-remove (keyword slug))
          id)
        (:parameters dashboard))))
```